.buildkite/scripts/cloud-cleanup-oidc.sh

```diff
@@ -0,0 +1,229 @@
+    if [ "${number}" -eq 0 ]; then
+        return 1
+    fi
+    return 0
+}
+
+# As long as cloud reaper does not support OIDC authentication.
+create_aws_ephemeral_user() {
+    # Generate a unique name for the ephemeral IAM user.
+    EPHEMERAL_USER="ephemeral-admin-$(date +%s)"
+    echo "Creating IAM user: ${EPHEMERAL_USER}"
+    aws iam create-user --user-name "${EPHEMERAL_USER}" \
+        --tags Key=ephemeral,Value=true Key=division,Value=engineering Key=org,Value=obs Key=environment,Value=ci Key=repo,Value=elastic-package Key=created_at,Value="$(date -Is)"
+
+    echo "Attaching AdministratorAccess policy to ${EPHEMERAL_USER}..."
+    aws iam attach-user-policy --user-name "${EPHEMERAL_USER}" --policy-arn arn:aws:iam::aws:policy/AdministratorAccess
+
+    echo "Creating access keys for ${EPHEMERAL_USER}..."
+    creds_json=$(aws iam create-access-key --user-name "${EPHEMERAL_USER}")
+    AWS_ACCESS_KEY_ID_EPHEMERAL=$(echo "$creds_json" | jq -r '.AccessKey.AccessKeyId')
+    AWS_SECRET_ACCESS_KEY_EPHEMERAL=$(echo "$creds_json" | jq -r '.AccessKey.SecretAccessKey')
+    export EPHEMERAL_USER AWS_ACCESS_KEY_ID_EPHEMERAL AWS_SECRET_ACCESS_KEY_EPHEMERAL
+}
+
+# Define cleanup function to delete the ephemeral IAM user regardless of script outcome.
+cleanup_ephemeral_user() {
+    echo "Cleaning up ephemeral IAM user: ${EPHEMERAL_USER}"
+    aws iam detach-user-policy --user-name "${EPHEMERAL_USER}" --policy-arn arn:aws:iam::aws:policy/AdministratorAccess
+    key_id=$(echo "$creds_json" | jq -r '.AccessKey.AccessKeyId')
+    aws iam delete-access-key --user-name "${EPHEMERAL_USER}" --access-key-id "${key_id}"
+    aws iam delete-user --user-name "${EPHEMERAL_USER}"
+    echo "Ephemeral IAM user ${EPHEMERAL_USER} deleted."
+}
+trap cleanup_ephemeral_user EXIT
+
+cloud_reaper_aws() {
+    echo "--- Configuring ephemeral user"
+    create_aws_ephemeral_user
+
+    echo "Validating configuration"
+    docker run --rm -v "$(pwd)/.buildkite/configs/cleanup.aws.yml":/etc/cloud-reaper/config.yml \
+      -e AWS_ACCESS_KEY_ID="$AWS_ACCESS_KEY_ID_EPHEMERAL" \
+      -e AWS_SECRET_ACCESS_KEY="$AWS_SECRET_ACCESS_KEY_EPHEMERAL" \
+      -e ACCOUNT_PROJECT="observability-ci" \
+      -e CREATION_DATE="${DELETE_RESOURCES_BEFORE_DATE}" \
+      "${CLOUD_REAPER_IMAGE}" \
+        cloud-reaper \
+          --debug \
+          --config /etc/cloud-reaper/config.yml \
+          validate
+
+    echo "Scanning resources"
+    docker run --rm -v "$(pwd)/.buildkite/configs/cleanup.aws.yml":/etc/cloud-reaper/config.yml \
+      -e AWS_ACCESS_KEY_ID="$AWS_ACCESS_KEY_ID_EPHEMERAL" \
+      -e AWS_SECRET_ACCESS_KEY="$AWS_SECRET_ACCESS_KEY_EPHEMERAL" \
+      -e ACCOUNT_PROJECT="observability-ci" \
+      -e CREATION_DATE="${DELETE_RESOURCES_BEFORE_DATE}" \
+      "${CLOUD_REAPER_IMAGE}" \
+        cloud-reaper \
+          --config /etc/cloud-reaper/config.yml \
+          ${COMMAND} | tee "${AWS_RESOURCES_FILE}"
+}
+
+echo "--- Installing awscli"
+with_aws_cli
+
```

Comment:
Run earlier, to help with using the aws cli to create the ephemeral service account